### DISCUSSION TITLE
Can't get collision to work with animated character

### DISCUSSION THREAD

**Discussion by whuang-blackline** [replit/kaboom#443] [Q&A] [answered]
Hi there,

I am working off some tutorials and can't get objects to repect collisions.  Here is the code.

kaboom.js
```
const k = kaboom(kaboomSettings)

export default k
```

main.js
```
k.loadSpriteAtlas('./sprites/faune_animation.png', {
    'faune': {
        x: 0,
        y: 0,
        width: 768,
        height: 32,
        sliceX: 24,
        anims: {
            'walk-down': { from: 0, to: 7, loop: true, speed: 15 },
            'walk-side': { from: 8, to: 15, loop: true, speed: 15 },
            'walk-up': { from: 16, to: 23, loop: true, speed: 15 },
            // NOTE: onEnd error when loop not set even though we don't need
            // to loop for a 1 frame animation
            'idle-up': { from: 19, to: 19, loop: true },
            'idle-side': { from: 10, to: 10, loop: true },
            'idle-down': { from: 1, to: 1, loop: true },
        }
    }
})

k.loadSprite('stone', './sprites/stone.png')
```

scene.js
```
    k.add([
        k.pos(width() * 0.25, height() * 0.25),
        k.sprite('stone'),
        k.origin('center'),
        k.area(),
        k.solid()
    ])

    // add player
    const player = k.add([
        k.pos(width() * 0.5, height() * 0.5),
        k.sprite('faune'),
        k.origin('center'),
        k.area(),
        k.solid()
    ])
```

The 'faune' characters passes through the stone.  Expected result is character will not pass through object.  Any help is appreciated.  Thanks!

**Comment by lajbel** [comment]
Ohh is strange, what version are you using?

**Reply by whuang-blackline** [reply]
```
$ yarn list kaboom
yarn list v1.22.17
warning Filtering by arguments is deprecated. Please use the pattern option instead.
└─ kaboom@2000.1.6
Done in 0.55s.
```

**Comment by slmjkdbtl** [comment]
Can you check if the object has the correct collider in inspect mode? (Press `F1` or `debug.inspect = true`)

**Comment by whuang-blackline** [comment]
Here is a screen shot of both objects.  Tried it using loadSprite and loadSpriteAtlas using anims property.

![Screenshot from 2021-11-21 18-00-18](https://user-images.githubusercontent.com/80294025/142786718-0fda4864-a160-4287-9752-5cef712b7507.png)

![Screenshot from 2021-11-21 18-00-28](https://user-images.githubusercontent.com/80294025/142786720-4dd0c717-9570-4daa-884d-189ba9305e9b.png

![Screenshot from 2021-11-21 18-01-40](https://user-images.githubusercontent.com/80294025/142786775-297a58de-0adc-4b78-a19d-4d15e89322f9.png)

**Comment by slmjkdbtl** [comment, marked as answer]
The mechanism to prevent solid objects moving into each other is inside `.move()` from `pos()` component (it needs to know the velocity to prevent tunneling), if you're using directly assigning `.pos = ` it won't stop them from moving into each other. 

If you want to directly assign `.pos` you can also call `.pushOutAll()` every frame to push the object out of every other solid objects, but that won't prevent tunneling when it's moving in high speed.

**Reply by whuang-blackline** [reply]
Thanks I will try it out !
